db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/JoinNode.java

```diff
@@ -1516,6 +1543,24 @@ private void deferredBindExpressions(FromList fromListParam) throws StandardExce
 
 		/* ON clause */
         if(joinClause!=null){
+            /* JoinNode.deferredBindExpressions() may be called again after the outer join rewrite
+               optimization, at this stage, we don't want to simplify the ON clause predicate again, especially
+               the top AND node with ParTrue
```

Comment:
I'm not sure if I completely understand this comment. 
This function deferredBindExpressions() is mainly called at the bind phase. But it could also be triggered during preprocess() phase after the outerjoins have gone through some transformations. We only want the constant folding and predicate simplification logic to be triggered for the bind phase call, and not the preprocess phase call, as it will simplify the predicates again and wipe out the top level ParTrue nodes added earlier by the logic JoinNode.normExpressions() during the preprocess() phase . Optimizer expect all the predicates to have a top level AND with ParTrue as right during the optimization phase.

Reply:
I'm just saying change "ParTrue" to some other description.  ParTrue is a Teradata thing!  :smiley_cat:

Reply:
I see. I'll make the change.